tests/unittest/noctx.hpp

```diff
@@ -0,0 +1,224 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(c) 2025 Intel Corporation
+ */
+
+#include <gtest/gtest.h>
+
+#include <map>
+#include <memory>
+#include <thread>
+#include <vector>
+
+#include "tests.hpp"
+
+#define AUDIO_CLOCK_HRTZ 48000
+#define VIDEO_CLOCK_HRTZ 90000
+#define SESSION_SKIP_PORT -1
+
+class Session;
+class SessionUserData;
+class NoCtxTest;
+class St30pHandler;
+class Handlers;
+
+/* Session class represents a media session that can run multiple threads */
+class Session {
+ public:
+  /* can be more than one thread */
+  std::vector<std::thread> threads;
+  /* shoudnt be more than one */
+  std::atomic<bool> stopFlag{false};
+
+  void addThread(std::function<void(std::atomic<bool>&)> func) {
+    threads.emplace_back(func, std::ref(stopFlag));
+  }
+
+  bool isRunning() const {
+    for (const auto& thread : threads) {
+      if (!thread.joinable()) {
+        return false;
+      }
+    }
+
+    return !threads.empty();
+  }
+
+  void stop() {
+    stopFlag = true;
+    for (auto& thread : threads) {
+      if (thread.joinable()) {
+        thread.join();
+      }
+    }
+    threads.clear();
```

Comment:
I think it would be much easier if you use std::jthread. Use std::stop_token in the thread and here just call threads.clear(); everything will be cleared automatically.